server/src/main/java/org/opensearch/action/get/TransportMultiGetAction.java

```diff
@@ -75,6 +78,28 @@ public TransportMultiGetAction(
         this.indexNameExpressionResolver = resolver;
     }
 
+    /**
+     * Returns true if MultiGet request should be routed to primary shards, else false.
+     */
+    boolean isPrimaryBasedRouting(MultiGetRequest request, MultiGetRequest.Item item) {
```

Comment:
nit - This is largely identical to the other method, can we extract this to a common place and pass in the item/realtime params?  Or at least the SR enabled check?

Reply:
Tried to look into this, but only common base class is TransportAction<Request, Response> which doesn't have access to clusterService which I don't think we should be passing in as an argument.